src/realm-settings/key-providers/aes-generated/AESGeneratedForm.tsx

```diff
@@ -0,0 +1,363 @@
+import { ViewHeader } from "../../../components/view-header/ViewHeader";
+import { convertToFormValues } from "../../../util";
+import { useAlerts } from "../../../components/alert/Alerts";
+
+type AESGeneratedFormProps = {
+  handleModalToggle?: () => void;
+  refresh?: () => void;
+  editMode?: boolean;
+  providerType: string;
+};
+
+export interface MatchParams {
+  providerType: string;
+}
+
+export const AESGeneratedForm = ({
+  editMode,
+  providerType,
+  handleModalToggle,
+  refresh,
+}: AESGeneratedFormProps) => {
+  const { t } = useTranslation("groups");
+  const serverInfo = useServerInfo();
+  const [isKeySizeDropdownOpen, setIsKeySizeDropdownOpen] = useState(false);
+
+  const adminClient = useAdminClient();
+  const { addAlert } = useAlerts();
+
+  const { id } = useParams<{ id: string }>();
+
+  const providerId =
+    useRouteMatch<MatchParams>("/:providerType?")?.params.providerType;
+
+  const save = async (component: ComponentRepresentation) => {
+    try {
+      if (id) {
+        await adminClient.components.update(
+          { id },
+          {
+            ...component,
+            parentId: component.parentId,
+            providerId: providerType,
+            providerType: "org.keycloak.keys.KeyProvider",
+          }
+        );
+        addAlert(t("realm-settings:saveProviderSuccess"), AlertVariant.success);
+      } else {
+        await adminClient.components.create({
+          ...component,
+          parentId: component.parentId,
+          providerId: providerType,
+          providerType: "org.keycloak.keys.KeyProvider",
+        });
+        handleModalToggle!();
+        addAlert(t("realm-settings:saveProviderSuccess"), AlertVariant.success);
+        refresh!();
+      }
+    } catch (error) {
+      addAlert(
+        t("realm-settings:saveProviderError", {
+          error: error.response?.data?.errorMessage || error,
+        }),
+        AlertVariant.danger
+      );
+    }
+  };
+
+  const form = useForm<ComponentRepresentation>({ mode: "onChange" });
+
+  const setupForm = (component: ComponentRepresentation) => {
+    form.reset();
+    Object.entries(component).map(([key, value]) => {
+      if (
+        key === "config" &&
+        component.config!.secretSize &&
+        component.config!.active
+      ) {
+        form.setValue("config.secretSize", value.secretSize[0]);
+
+        form.setValue("config.active", value.active[0]);
+
+        convertToFormValues(value, "config", form.setValue);
+      }
+      form.setValue(key, value);
+    });
+  };
+
+  useFetch(
+    async () => {
+      if (editMode) return await adminClient.components.findOne({ id: id });
+    },
+    (result) => {
+      if (result) {
+        setupForm(result);
+      }
+    },
+    []
+  );
+
+  const allComponentTypes =
+    serverInfo.componentTypes?.["org.keycloak.keys.KeyProvider"] ?? [];
+
+  const AESSecretSizeOptions = allComponentTypes[0].properties[3].options;
```

Comment:
It's conventional to use the camelCase convention for variables that are not global constants.
```suggestion
  const aesSecretSizeOptions = allComponentTypes[0].properties[3].options;
```